portfolio/src/test/java/com/google/sps/servlets/CommentsServletTest.java

```diff
@@ -26,16 +43,127 @@ public void setUp() {
   }
 
   @Test
-  void testPost() {
-    Gson gson = new Gson();
-    Comment comment = new Comment();
+  public void testCreateWithValidPayload() {
+    request.setMethod("POST");
+    request.setContentType("text/html");
+    request.addParameter("username", "Paul");
+    request.addParameter("text", "A nice comment");
+    try {
+      servlet.doPost(request, response);
+      assertEquals(response.getStatus(), REDIRECT);
+      String responseString = response.getContentAsString();
+      JsonObject jsonObject = (new JsonParser()).parse(responseString).getAsJsonObject();
+      assertEquals(jsonObject.get("username").toString(), quoted("Paul"));
+      assertEquals(jsonObject.get("text").toString(), quoted("A nice comment"));
+    } catch (IOException ex) {
+      System.out.println(ex.getMessage());
+    }
+  }
+
+  @Test
+  public void testCreateWithValidPayloadAndValidReplyKey() {
+    // create a comment and put it to the datastore
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    Entity commentEntity = new Entity("Comment");
+    ds.put(commentEntity);
+    commentEntity.setProperty("username", "paul");
+    commentEntity.setProperty("text", "Very cool comment text");
+    request.setMethod("POST");
+    request.setContentType("text/html");
+    request.addParameter("username", "Paul");
+    request.addParameter("text", "A nice comment");
+    // add a valid reply to field
+    String replyToKey = KeyFactory.keyToString(commentEntity.getKey());
+    request.addParameter("replyTo", replyToKey);
+    try {
+      servlet.doPost(request, response);
+      assertEquals(response.getStatus(), REDIRECT);
+      String responseString = response.getContentAsString();
+      JsonObject jsonObject = (new JsonParser()).parse(responseString).getAsJsonObject();
+      assertEquals(jsonObject.get("username").toString(), quoted("Paul"));
+      assertEquals(jsonObject.get("text").toString(), quoted("A nice comment"));
+      assertEquals(jsonObject.get("replyTo").toString(), quoted(replyToKey));
+    } catch (IOException ex) {
+      System.out.println(ex.getMessage());
+    }
+  }
+
+  @Test
+  public void testCreateWithNoPayload() {
+    request.setMethod("POST");
+    request.setContentType("text/html");
+    try {
+      servlet.doPost(request, response);
+      assertEquals(response.getStatus(), BAD_REQUEST);
+    } catch (IOException ex) {
+      System.out.println(ex.getMessage());
+    }
+  }
+
+  @Test
+  public void testCreateWithAbsentCommentText() {
+    request.setMethod("POST");
+    request.setContentType("text/html");
+    // set the username only, without the text
+    request.addParameter("username", "Paul");
+    try {
+      servlet.doPost(request, response);
+      assertEquals(response.getStatus(), BAD_REQUEST);
+    } catch (IOException ex) {
+      System.out.println(ex.getMessage());
+    }
+  }
+
+  @Test
+  public void testCreateWithInvalidCommentTextValue() {
```

Comment:
suggest making these tests named testCreateCommentFailInvalidCommentTextValue() and so on

Reply:
You mean I have to specify the expected condition like fail/succes?

Reply:
Yes. the current name seems to mean like you can create with invalid comment text value which is not the case.